### PR TITLE
refactor(view): rename collection template data to models

### PR DESCRIPTION
**Pull request by paulfalgout**
## Linked issue

No linked issue; this resolves the serialized-collection decision in the v5 API-shape freeze roadmap.

## What

- Rename collection template data from `items` to `models` in the shared View/CollectionView rendering path and generated distributions.
- Add source and built-format coverage proving `models` is present, `items` is absent, and custom `serializeCollection()` results remain supported.
- Update the roadmap decision, terminology audit, rendering docs, changelog, upgrade guide, and v4 compatibility ledger.

## Why

The pre-stable template property conflicted with v5's canonical model vocabulary while the selected data contract exposes `DataApi.models(collection)`. Resolving that split before the API freeze leaves one term for collection records. The documentation explicitly distinguishes raw model instances returned by `DataApi.models()` from the serialized values supplied to templates.

## Agent-development failure addressed

- Removes the remaining `items`/`models` vocabulary split that required agents to memorize a one-off template exception.
- Keeps the raw-versus-serialized shape distinction explicit instead of relying on an undocumented naming convention.

## Public behavior contract

- Canonical behavior after this PR: a View or CollectionView with a collection and no model passes `{ models: serializeCollectionResult }` to its template; the default result is an array of serialized values, while an override may return another shape.
- Superseded behavior removed, if any: the pre-stable `items` template property is removed.
- Compatibility exception and removal condition, if any: none; there is no alias or dual-name fallback.

## Scope

- Affects collection serialization for View and CollectionView templates, their generated ESM/CJS/UMD distributions, tests, and migration documentation.
- Does not change `serializeCollection()` output, DataApi behavior, child-view reconciliation, or model-template serialization.

## Runtime cost

- [ ] Static or documentation only
- [ ] Development or test only; excluded from production entrypoints
- [x] Changes an existing production path
- [ ] Adds an explicitly opt-in runtime path

Measured impact or explanation:

- Bundle: exact-base cumulative Brotli is 89,831 B versus 89,856 B, a 25 B reduction (-0.028%). No artifact crosses the 1% approval threshold.
- Startup/hot path: one object-property name changes during collection template serialization; no new operation is added.
- Allocations/retention: exact-base resource measurements are identical.

## Deployment Impact

No deployment or redeploy is required beyond normal publication of a future Marionette version. This changes the pre-stable package API only.

## Validation

- [x] Tests cover the acceptance criteria and relevant edge cases.
- [x] Docs, examples, declarations, diagnostics, and rule metadata agree where relevant.
- [x] No private consumer, private fixture, or undocumented context is required.
- [x] I ran the commands listed below and am reporting their actual results.

Commands:

```sh
npm test -- --reporter=dot
npm run lint:ci
npm run build
npm run test:source
npm run test:dist
npm run test:fixtures
npm run test:browser
npm run docs:check
npm run test:performance
npm run size
```

Results: 110 test files and 2,049 tests passed; lint, source/distribution checks, the complete packed-package fixture matrix, Chromium/Firefox/WebKit browser tests, 22 executable documentation examples, 56 generated HTML files with 503 internal links, and 118 performance tests passed. A clean rebuild produced no distribution diff.

## Package and browser impact

- Entry points or install fixtures affected: the root package's ESM, CommonJS, unminified UMD, and minified UMD outputs; fixtures all pass.
- Browser contracts affected: collection template input naming only; the complete supported browser suite passes.
- Production/dev/test module-graph impact: no module is added or removed; only the shared template-render property changes.

## Rollback or deprecation

Revert this PR before v5 stabilization if the decision changes. Do not add an `items` compatibility alias.

## Review notes

A bounded Claude review challenged repository-sweep completeness and the raw-versus-serialized distinction. That led to updating the stale terminology audit, adding direct CollectionView and built-format tests, and clarifying the shape distinction. Its suggestion to use `serializedModels` was rejected because the settled roadmap intentionally standardizes the domain vocabulary on `models`.

---
## Summary by cubic
Renames the serialized collection template property from `items` to `models` for `View` and `CollectionView`, aligning template data with v5’s model vocabulary. The old property is removed without an alias; `serializeCollection()` output and `DataApi.models()` behavior remain unchanged.

**Migration**

- Replace collection-template reads and destructuring of `items` with `models`; no redeploy is required.

**Validation**

- Covers `View`, `CollectionView`, custom serialization, and generated ESM, CommonJS, and UMD distributions, including the absence of `items`.

<sup>Written for commit e33c97c13ce9733b9d306006ea1e739762f295cb. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/marionettejs/marionette/pull/392?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

- **Changed**
  - Collection data passed to view templates is now available under `models` instead of `items`.
  - The new property contains serialized model values, not raw model instances.
  - No compatibility alias is provided for the former `items` property.

- **Documentation**
  - Updated rendering guidance, terminology, roadmap, changelog, and migration instructions with the new property name and upgrade example.

- **Tests**
  - Added and updated coverage for empty collections, custom serialization, rendering behavior, and absence of the legacy property.

### CODE REVIEW

**Comment by paulfalgout** [comment]
**Approved** — verified at fetched SHA `b735972f` (base `4f74056c`). Core aggregate **88,375**, down 25 B.

### The API-shape gate worked end to end

This is the first item recorded under the **Required before API-shape freeze** category created in #386, and it closes cleanly:

```diff
- **Required before API-shape freeze:** Decide the serialized collection template
-   property that is currently named `items`. …
+ **Selected:** Rename the pre-stable serialized collection template property from
+   `items` to `models` without an alias …
```

The audit's `Unresolved inconsistency` row is gone too, replaced by `Intentional historical reference` entries for the migration docs and the negative assertion. Raised, tracked, decided, and reclassified in six PRs — which is the behaviour I was hoping for when I argued that "a later API decision may address it" was too weak. Worth noting the category earned its keep on its first use.

### The shape distinction is handled where it matters

Unifying the vocabulary creates a real hazard: `DataApi.models()` and the template `models` now share a word for different shapes. Both docs address it head-on, and the rendering doc does it best by naming the transform rather than gesturing at it:

> These are the results of calling `DataApi.serialize()` for each model, not the raw model instances returned by `DataApi.models()`.

That is checkable guidance. "Serialized values" alone would have left readers guessing which serialization.

### One small thing: the example reintroduces the ambiguity

```js
<% _.each(models, function(model) { %>
  <li><%- model.name %></li>
<% }) %>
```

Each element here is a serialized value, not a model — the prose two paragraphs up says exactly that. But the loop variable is named `model`, so the example teaches the reading the documentation just disclaimed. A reader who copies it and later reaches for `model.get('name')` or `model.cid` will find neither, and the example is what people copy.

`<% _.each(models, function(data) { %>` — or `attributes`, or `serialized` — would cost nothing and would make the sample agree with the sentence above it. Small, but this is the one place in the rename where the shared word can actually mislead, and it is the most-copied line in the section.

**Comment by paulfalgout** [comment]
Addressed in amended head `4b6f18974f7fb2f44053e76e2e07cac5cc3e0118`: the collection-template example now names each serialized entry `data`, matching the documented raw-model versus serialized-value distinction. `npm run docs:check` and `npm run lint:ci` pass on the amended head. Please refresh approval against this exact SHA.

**Comment by paulfalgout** [comment]
**Approval stands** at `4b6f1897`. Docs only; core aggregate remains **88,375**.

The example now agrees with the sentence above it — `data` is a serialized value and reads as one, so nobody copying it will reach for `.get()` or `.cid` on something that has neither. That was the last place in the rename where the shared `models` word could still mislead.

Everything I raised on this PR is addressed.

File: test/unit/mixins/template-render.spec.js [hunk 1]
```diff
@@ -252,10 +252,23 @@ describe('template-render', function() {
         expect(renderer.serializeCollection).to.be.calledOnce;
       });
 
-      it('should send collection data on the established `items` template property', function() {
+      it('should send collection data on the `models` template property', function() {
         expect(renderer._renderHtml)
           .to.be.calledOnce
-          .and.calledWith(renderer.template, { items: [{ id: 1 },{ id: 2 }] });
+          .and.calledWith(renderer.template, { models: [{ id: 1 },{ id: 2 }] });
+        expect(renderer._renderHtml.firstCall.args[1]).to.not.have.property('items');
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply test/unit/mixins/template-render.spec.js]: Addressed in amended head `b1b969ec8b27d67c38ed5290e64ff65754130c80`: the new contract and custom-serialization assertions now inspect the public template callback argument rather than `_renderHtml` call history. The existing plain-View and CollectionView integration tests continue to cover rendered behavior. `npm test -- test/unit/mixins/template-render.spec.js --reporter=dot` (29 tests) and `npm run lint:ci` pass.

**Comment by paulfalgout** [comment]
The CodeRabbit thread is fixed and resolved at exact head `b1b969ec8b27d67c38ed5290e64ff65754130c80`. The new assertions now use the public template callback; focused tests and lint pass. Please refresh approval against this exact SHA.

**Comment by paulfalgout** [comment]
**Approval stands** at `b1b969ec`. Test-only; core aggregate remains **88,375**.

Moving these assertions from `_renderHtml` to a `template` spy tightens what they actually prove. The PR's stated contract is that a View with a collection *"passes `{ models: serializedValues }` to its template"* — asserting on the template function proves that sentence directly, where asserting on `_renderHtml(template, data)` proved only that the renderer passes the data one step toward it, through a private method whose name and signature are not part of the contract.

It also makes the negative assertion stronger in the same move: `renderer.template.firstCall.args[0]` has no `items` property is a statement about what a consumer's template receives, which is where the absence needs to hold.

Consistent with the private-internals cleanup in #381's reconciliation specs — same direction, and worth the churn for the same reason.

File: upgradeGuide.md [hunk 2]
```diff
@@ -19,6 +19,17 @@ current public behavior boundary. Final migration documentation is tracked in
   // v5
   const models = DataApi.models(collection);
   ```
+- View templates with a collection now receive its serialized records on the
+  `models` property. Replace the pre-stable `items` property without retaining both
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply upgradeGuide.md]: Addressed in amended head `e33c97c13ce9733b9d306006ea1e739762f295cb`: the upgrade guide now scopes the property to a View with a collection and no model, names the `serializeCollection()` result, and describes only the default as an array of serialized values. The PR public-contract text now uses the same qualification. `npm run docs:check` and `npm run lint:ci` pass.

File: docs/terminology.md [hunk 3]
```diff
@@ -18,8 +18,8 @@ is necessary to explain a real migration.
...
 
-Public API names override prose preferences when documenting a retained contract. In
-particular, templates still receive serialized collections on the `items` property,
-while collection adapters expose `DataApi.models(collection)`. The API-shape gate
-must resolve that split before stable v5. Code that must prove a removed API may name
-it in a negative assertion.
+Public API names override prose preferences when documenting a retained contract.
+The shared `models` vocabulary does not erase shape distinctions: `DataApi.models()`
+returns raw model instances, while a collection template receives each model's
+serialized value.
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply docs/terminology.md]: Addressed in amended head `e33c97c13ce9733b9d306006ea1e739762f295cb`: terminology now says `DataApi.models()` returns the source raw ordered snapshot, default `serializeCollection()` returns per-model serialized values, an override may return another shape, and template `models` receives that result only when no model is present. The compatibility ledger matches. `npm run docs:check` and `npm run lint:ci` pass.

**Comment by paulfalgout** [comment]
Both late Cubic documentation threads are fixed and resolved at exact head `e33c97c13ce9733b9d306006ea1e739762f295cb`. The docs and PR contract now state model precedence, the raw DataApi snapshot, the default serialized-value array, and the arbitrary override-result shape precisely. Docs and lint pass. Please refresh approval against this exact SHA.

**Comment by paulfalgout** [comment]
**Approval stands** at `e33c97c1`. Docs only; core aggregate remains **88,375**.

Both precisions here fix text I read and passed, so worth naming what they actually correct.

**"with a collection and no model."** `serializeData()` returns `serializeModel()` whenever `this.model` is set and only falls through to the collection branch otherwise — so a View with both never sees `models` at all. All three documents previously said "a View with a collection," which is true of the condition but silent on the precedence. A reader with both configured would have expected a `models` property that never arrives, and nothing in the old wording would have told them why.

**"an override may return another shape."** This is the sharper of the two. The previous upgrade-guide text promised *"its serialized records"* and I described it as an array of serialized values without qualification. `serializeCollection()` is overridable, and your own test pins a non-array return:

```js
renderer.serializeCollection = this.sinon.stub().returns({ custom: true });
… .and.calledWith({ models: serialized });
```

So `models` is contractually *the result of `serializeCollection()`* — an array only by default. Separating the default from the guarantee is right, because a consumer who overrode serialization and read the old sentence would reasonably have concluded core would wrap or normalize their result. It does not; it passes it through.

Tightening `DataApi.models()` to "the source's raw ordered model snapshot" also matches the vocabulary settled in #386 rather than paraphrasing it.

I had called the earlier raw-versus-serialized wording "handled head-on," and it was — on the distinction it addressed. These two conditions sat outside it, and both are the kind of thing that surfaces as "why is my template data undefined" rather than as an error.
